src/modules/navigator/mission_feasibility_checker.cpp

```diff
@@ -275,31 +289,93 @@ bool MissionFeasibilityChecker::checkMissionItemValidity(dm_item_t dm_current, s
 		    missionitem.nav_cmd != NAV_CMD_DO_SET_CAM_TRIGG_DIST &&
 		    missionitem.nav_cmd != NAV_CMD_DO_VTOL_TRANSITION) {
 
-			mavlink_log_critical(_mavlink_log_pub, "Rejecting mission item %i: unsupported cmd: %d", (int)(i + 1),
+			mavlink_log_critical(_navigator->get_mavlink_log_pub(), "Mission rejected: item %i: unsupported cmd: %d", (int)(i + 1),
 					     (int)missionitem.nav_cmd);
 			return false;
 		}
 
+		/* Check non navigation item */
+		if (missionitem.nav_cmd == NAV_CMD_DO_SET_SERVO) {
+
+			/* check actuator number */
+			if (missionitem.params[0] < 0 || missionitem.params[0] > 5) {
+				mavlink_log_critical(_navigator->get_mavlink_log_pub(), "Actuator number %d is out of bounds 0..5",
+						     (int)missionitem.params[0]);
+				return false;
+			}
+
+			/* check actuator value */
+			if (missionitem.params[1] < -2000 || missionitem.params[1] > 2000) {
```

Comment:
Yes, someone through this in the distance to waypoint 1 check. I can fix it while I'm here.